### PR TITLE
refactor(build): Improve the project CMake

### PR DESCRIPTION
**Pull request by acelyc111**
This patch includes these changes:
- Move `cmake_minimum_required` to the top of CMakeLists.txt files.
- Do not pass useless cmake options when build thirdparties.
- Generate a make parallel option to speed up building thirdparty libraries.
- Add `DOWNLOAD_EXTRACT_TIMESTAMP` and `DOWNLOAD_NO_PROGRESS` when build thirdparties
  (See https://cmake.org/cmake/help/latest/module/ExternalProject.html).
- Remove the useless `gflags` thirdparty library.
- Add a patch from offcial repository to build `rocksdb`, so it's possible to build
  thirdparty directly in `thirdparty` diretory.

### CODE REVIEW

File: .licenserc.yaml [hunk 1]
```diff
@@ -69,9 +69,9 @@ header:
     - 'thirdparty/fix_jemalloc_for_m1_on_macos.patch'
     - 'thirdparty/fix_libevent_for_macos.patch'
     - 'thirdparty/fix_prometheus-cpp_limits.patch'
+    - 'thirdparty/fix_rocksdb-cmake-PORTABLE-option.patch'
     - 'thirdparty/fix_s2_for_aarch64.patch'
     - 'thirdparty/fix_thrift_for_cpp11.patch'
-    - 'thirdparty/rocksdb_fix_atomic_flush_0879c240.patch'
```
Comment (empiredan) [inline .licenserc.yaml:74]: Is `thirdparty/rocksdb_fix_atomic_flush_0879c240.patch` not needed ? Could this file be removed ?

Comment (acelyc111) [reply .licenserc.yaml:74]: It's not needed after we bump rocksdb to 8.5.3, I forget to remove it.

Comment (empiredan) [reply .licenserc.yaml:74]: OK, could we just remove it in this PR ? `Check License` also said that `thirdparty/rocksdb_fix_atomic_flush_0879c240.patch` should be removed once it's not in `.licenserc.yaml`:

```
INFO Loading configuration from file: .licenserc.yaml 
INFO Totally checked 1705 files, valid: 1065, invalid: 1, ignored: 639, fixed: 0 
ERROR the following files don't have a valid license header: 
thirdparty/rocksdb_fix_atomic_flush_0879c240.patch 
ERROR one or more files does not have a valid license header 
Error: Process completed with exit code 1.
```

Comment (acelyc111) [reply .licenserc.yaml:74]: Oh, I removed it just now.

**Review by empiredan — APPROVED** [review]

**Review by Smityz — APPROVED** [review]

**Pull request merged**